packages/source-drupal/src/DrupalSource.js

```diff
@@ -0,0 +1,201 @@
+const axios = require('axios')
+const reduce = require('lodash.reduce')
+const { 
+  cullByWordCount
+} = require('./utils')
+const slugify = require('@sindresorhus/slugify')
+const Nodes = require('./entities/Nodes')
+const TaxonomyTerms = require('./entities/TaxonomyTerms')
+const Files = require('./entities/Files')
+const Users = require('./entities/Users')
+
+
+class DrupalSource {
+  // defaultOptions merged with this.options in App.vue
+  static defaultOptions() {
+    return {
+      baseUrl: '',
+      views: [], // deprecated
+      entities: [],
+      format: 'json'
+    }
+  }
+
+  constructor(api, options = {}) {
+    if (!options.baseUrl) throw new Error('baseUrl option is required')
+
+    this.options = options
+    this.supportedEntities = {
+      node: Nodes,
+      taxonomy_term: TaxonomyTerms,
+      file: Files,
+      user: Users,
+    }
+    this.entities = {}
+    this.apiSchema = {}
+
+    api.loadSource(store => this.initialize(store))
+  }
+
+  async initialize(store) {
+    try {
+      this.store = store
+      this.apiSchema = await this.fetchJsonApiSchema()
+
+      await this.processEntities()
+    } catch (error) {
+      let { message } = error
+      throw new Error(message)
+    }
+  }
+
+  async fetchJsonApiSchema() {
+    const { baseUrl } = this.options
```

Comment:
```suggestion
    const { baseUrl, apiUrl } = this.options
```

Reply:
This would be a nice option. Maybe `apiBase` is a better option name since it's not a full URL?

Reply:
I like `apiBase` as well - I'll get that in place.

What is the preferred way to do this?  I can't edit the suggestion made by @tanc, but I can provide a new suggestion which forces me to start a "review". Is it easiest for me just to resolve the conversation and push new code? Or @tanc can you edit your own suggestion and then I can just batch these suggestions into a commit?